File: src/main/java/axirassa/webapp/components/AxTextField.scala

package axirassa.webapp.components

import org.apache.tapestry5.annotations.Component
import org.apache.tapestry5.annotations.Parameter
import org.apache.tapestry5.annotations.Property
import org.apache.tapestry5.corelib.components.TextField

class AxTextField extends AxAbstractTextInput {
  @Component(
    publishParameters = "annotationProvider,clientId,disabled,nulls,translate,value",
    parameters = Array("validate=inherit:validate"))
  var txtfield : TextField = _

  @Parameter(defaultPrefix = "validate")
  @Property
  var validate : String = _

  def getTextField = txtfield

  override def getControlName = txtfield.getControlName

  override def isDisabled = txtfield.isDisabled

  override def isRequired = txtfield.isRequired

  override def getClientId = txtfield.getClientId

}

File: src/main/java/axirassa/webapp/components/AxTextField.java

package axirassa.webapp.components;

import org.apache.tapestry5.ComponentResources;
import org.apache.tapestry5.Field;
import org.apache.tapestry5.annotations.Component;
import org.apache.tapestry5.annotations.Property;
import org.apache.tapestry5.corelib.components.TextField;
import org.apache.tapestry5.ioc.annotations.Inject;
import org.apache.tapestry5.services.javascript.JavaScriptSupport;

public class AxTextField implements Field {
	@Inject
	private ComponentResources componentResources;

	@Inject
	private JavaScriptSupport jsSupport;

	@Component(publishParameters = "label,annotationProvider,clientId,disabled,nulls,translate,value,id,validate")
	@Property
	private TextField textfield;

	private String clientId;


	public void setupRender() {
		clientId = jsSupport.allocateClientId(componentResources);
	}


	@Override
	public String getControlName() {
		return textfield.getControlName();
	}


	@Override
	public boolean isDisabled() {
		return textfield.isDisabled();
	}


	@Override
	public boolean isRequired() {
		return textfield.isRequired();
	}


	@Override
	public String getClientId() {
		return clientId;
	}


	@Override
	public String getLabel() {
		return textfield.getLabel();
	}
}
